### PR TITLE
update-email: email change does not affect the username when "Email as username" option is checked

### LINKED ISSUE #13988: 19 - update-email feature - email change does not affect the username when "Email as username" option is checked
### Describe the bug

Hello,
I am testing the new update_email feature.
If I have the option "Email as username" active when changing email, it changes the email but not its username.
Is there any other option so that the email change affects the username when the "Email as username" option is checked ?
Thanks.

### Version

19

### Expected behavior

If the login option "Email as username" is active, when I change the email it must also change the username.

### Actual behavior

If the login option "Email as username" is active, when I change the email I have the old email as username.

### How to Reproduce?

1. activate the update-email feature
2. have the login option "email as username" activated
3. then log in with a user 
4. go to "personnal info" 
5. and modify his email 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by reda-alaoui**
Fixes #13988

### CODE REVIEW

**Comment by reda-alaoui** [comment]
@mposolda could somebody take a look at this? :)

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AbstractAppInitiatedActionUpdateEmailTest.java [hunk 1]
```diff
@@ -57,6 +58,13 @@ public void beforeTest() {
 				.lastName("Doh").build();
 		prepareUser(user);
 		ApiUtil.createUserAndResetPasswordWithAdminClient(testRealm(), user, "password");
+		setRegistrationEmailAsUsername(testRealm(), false);
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AbstractAppInitiatedActionUpdateEmailTest.java:61]: Nitpick: Is it possible to remove the change on line  61 and instead call `setRegistrationEmailAsUsername(false);` at the end of the `updateWithEmailAsUsernameEnabled()` test? This is just to make sure that test is not affected if default settings for the realm are changed (For example if emailAsUsername is ever changed to true in the realm).

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AbstractRequiredActionUpdateEmailTest.java [hunk 2]
```diff
@@ -72,9 +73,16 @@ public void beforeTest() {
 				.requiredAction(UserModel.RequiredAction.UPDATE_EMAIL.name()).build();
 		prepareUser(user);
 		ApiUtil.createUserAndResetPasswordWithAdminClient(testRealm(), user, "password");
+		setRegistrationEmailAsUsername(testRealm(), false);
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AbstractRequiredActionUpdateEmailTest.java:76]: Nitpick: Is it possible to move this to the end of new test method instead? See similar comment for `AbstractAppInitiatedActionUpdateEmailTest`.

**Review by mposolda — CHANGES_REQUESTED** [review]
@reda-alaoui Thanks for this fix!

I've added 2 minor comments inline. Could you please check them?

**Comment by reda-alaoui** [comment]
> @reda-alaoui Thanks for this fix!
> 
> I've added 2 minor comments inline. Could you please check them?

Done :+1:

**Review by pedroigor — APPROVED** [review]

**Review by mposolda — APPROVED** [review]
@reda-alaoui @pedroigor Thanks for the update and review!

**Pull request merged**